pinot-ingestion-jobs/pinot-ingestion-common/src/main/java/org/apache/pinot/ingestion/common/DefaultControllerRestApi.java

```diff
@@ -121,7 +122,47 @@ public void pushSegments(FileSystem fileSystem, List<Path> tarFilePaths) {
             LOGGER.error("Caught exception while pushing segment: {} to location: {}, retry {}/{}", segmentName,
                 pushLocation, retry, _retry, e);
             if (retry == _retry) {
-              throw new RuntimeException(String.format("Failed to push segment %s to %s with %d retries",segmentName, pushLocation, retry), e);
+              throw new RuntimeException(
+                  String.format("Failed to push segment %s to %s with %d retries", segmentName, pushLocation, retry),
+                  e);
+            }
+            try {
+              // Exponential back-off, max sleep time is 64 seconds.
+              Thread.sleep(1000 * (int) Math.pow(2, Math.min(retry + 1, 6)));
```

Comment:
Wny not use RetryPolicies? Or, are we not using this because of dependencies? If so, please add a TODO

Reply:
Will remove the dependency from `pinot-ingestion-common` from `pinot-standalone`